File: Iniciante/1099 - Soma de Ímpares Consecutivos II/TypeScript/1099.ts
import * as fs from "fs";

function main(): void {
  const input: number[] = fs.readFileSync(0, "utf8").trim().split(/\s+/).map(Number);

  let index: number = 0;
  const n: number = input[index++];

  for (let i = 0; i < n; i++) {
    const x: number = input[index++];
    const y: number = input[index++];

    console.log(calculateOddSum(x, y));
  }
}

function calculateOddSum(x: number, y: number): number {
  let sum: number = 0;
  let end: number = Math.max(x, y);
  let start: number = Math.min(x, y) + 1;

  if (start % 2 === 0) {
    start += 1;
  }

  for (let j = start; j < end; j += 2) {
    sum += j;
  }

  return sum;
}

main();

File: Iniciante/1099 - Soma de Ímpares Consecutivos II/Python/1099.py
def main():
    n = int(input())

    for i in range(n):
        x, y = map(int, input().split())
        print(calculateOddSum(x, y))


def calculateOddSum(x, y):
    total_sum = 0
    end = max(x, y)
    start = min(x, y) + 1

    if start % 2 == 0:
        start += 1

    j = start
    while j < end:
        total_sum += j
        j += 2

    return total_sum

if __name__ == "__main__":
    main()
